modules/reindex/src/main/java/org/elasticsearch/index/reindex/StartReindexJobAction.java

```diff
@@ -0,0 +1,158 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.CompositeIndicesRequest;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+public class StartReindexJobAction extends ActionType<StartReindexJobAction.Response> {
+
+    public static final StartReindexJobAction INSTANCE = new StartReindexJobAction();
+    // TODO: Name
+    public static final String NAME = "indices:data/write/start_reindex";
+
+    private StartReindexJobAction() {
+        super(NAME, Response::new);
+    }
+
+    public static class Request extends AcknowledgedRequest<Request> implements ToXContentObject, CompositeIndicesRequest {
+
+        private final ReindexRequest reindexRequest;
+        private final boolean waitForCompletion;
+
+
+        public Request(ReindexRequest reindexRequest) {
+            this(reindexRequest, false);
+        }
+
+        public Request(ReindexRequest reindexRequest, boolean waitForCompletion) {
+            this.reindexRequest = reindexRequest;
+            this.waitForCompletion = waitForCompletion;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            reindexRequest = new ReindexRequest(in);
+            waitForCompletion = in.readBoolean();
+        }
+
+        @Override
+        public void readFrom(StreamInput in) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            reindexRequest.writeTo(out);
+            out.writeBoolean(waitForCompletion);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) {
+            return builder;
+        }
+
+        public ReindexRequest getReindexRequest() {
+            return reindexRequest;
+        }
+
+        public boolean getWaitForCompletion() {
+            return waitForCompletion;
+        }
+    }
+
+    public static class Response extends AcknowledgedResponse {
```

Comment:
It seems we either return acknowledged=true or an exception. Could this just be an ActionResponse then?